### PR TITLE
Fixed PropertyModal

### PR DESCRIPTION
**Pull request by sarthak-marwaha**
Ticket: https://nanopay.atlassian.net/browse/NP-4605
Updated PropertyModal to extend StyledModal that subsequently fixes CSS issues

### CODE REVIEW

File: src/foam/u2/dialog/StyledModal.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ foam.CLASS({
             .start().addClasses(['h400', this.myClass('title')]).add(this.title).end()
             .start()
               .addClass(this.myClass('modal-body'))
-              .tag('', null, this.content$)
+              .add(this.addBody())
```
Comment (tharmaman) [inline src/foam/u2/dialog/StyledModal.js:149]: So you didn't end up  having to change content?

File: src/foam/u2/dialog/StyledModal.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ foam.CLASS({
 
   methods: [
     function init() {
-      bgColor = this.returnExpandedCSS(this.modalStyle.color);
```
Comment (tharmaman) [inline src/foam/u2/dialog/StyledModal.js:124]: Lol was this breaking before?

File: src/foam/u2/PropertyModal.js [hunk 3]
```diff
@@ -76,39 +53,33 @@ foam.CLASS({
...
-          .startContext({ data: this })
-            .tag(this.CANCEL, { buttonStyle: 'SECONDARY' })
-            .tag(this.OK)
-          .endContext()
-        .end();
+        .endContext();
+    },
+    function addActions() {
+      var actions = this.E().startContext({ data: this });
+      actions.tag(this.OK, { isDestructive: this.modalStyle == 'DESTRUCTIVE' });
```
Comment (tharmaman) [inline src/foam/u2/PropertyModal.js:71]: So default actions are OK and  CANCEL?

**Review by tharmaman — APPROVED** [review]
Looks good overall, just some confirmation questions

File: src/foam/u2/dialog/StyledModal.js (hunk 1 above)
Comment (sarthak-marwaha) [reply src/foam/u2/dialog/StyledModal.js:149]: No, had a talk with Kevin and he said this was the way to do it. Just changing content was causing DOM loading issues

File: src/foam/u2/dialog/StyledModal.js (hunk 2 above)
Comment (sarthak-marwaha) [reply src/foam/u2/dialog/StyledModal.js:124]: Nope, it was working perfectly lol

File: src/foam/u2/PropertyModal.js (hunk 3 above)
Comment (sarthak-marwaha) [reply src/foam/u2/PropertyModal.js:71]: Yeah and they can be overridden just like they were before

**Pull request merged**